Update product labels front matter documentation to focus only on cascading front matter (#234)
grafana/writers-toolkit: docs/sources/writing-guide/front-matter/index.md

Provides an HTML redirect from the pages in the list to the current page.
For more information, refer to [Hugo aliases](#hugo-aliases).

### date

Describes the initial publish date of the page.
Hugo produces XML page outputs for use by RSS feeds where users can be notified of updates.
Customers use RSS feeds of our release notes pages to be notified of new releases.

> Therefore, the `date` front matter is recommended for release note pages.

Reviewer: does this apply to what's new (we don't have Grafana release notes)

Author: I think we should start to put publish dates in our What's New pages, yes.

> The value to the `date` field should be a full ISO 8601 timestamp.

Reviewer: ```suggestion
The value of the `date` field should be a full ISO 8601 timestamp.
```

- `enterprise`: "Enterprise"
- `oss`: "Open source"

Labels are inherited through cascading front matter.
All labels, including project defaults, are set in the project's root index file.
This is almost always the `docs/sources/_index.md` file in the project repository.

If you add a new page, consider whether the default labels are appropriate.
If the default labels are incorrect for a page or directory of pages, update the cascading front matter in `docs/sources/_index.md`.

#### Set cascading front matter

> Set cascading front matter by adding or updating the `cascade` mapping in the `docs/sources/_index.md` front matter:

Reviewer 2: From here I suggest deleting everything from line 105-122. And then simply showing the example that starts with the intro at line 123. It's a much more efficient way to communicate to non-technical person what to do and exactly how.

```suggestion
Set cascading front matter by adding or updating the `cascade` mapping in the `docs/sources/_index.md` front matter.
```

```yaml
cascade: <TARGETS...>
```

Where `<TARGETS...>` is a sequence of mappings that looks similar to the following:

```yaml
- _target: { path: <PATH> }
  labels:
    products: <PRODUCTS...>
```

- `<PATH>` is a glob pattern that matches the affected source pages.
  For more information about the the `<PATH>` value, refer to [Specify a path](#specify-a-path).

- `<PRODUCTS...>` is a sequence of valid product labels.

The following example target mapping sets the product label to "Grafana Cloud" for all pages under `/docs/grafana/latest/troubleshooting/`:

```yaml
- _target: { path: /docs/grafana/*/troubleshooting/** }
  labels:
    products:
      - cloud
```

##### Specify a path

The value of the `_target.path` mapping is a string matching paths from the Hugo content root.
Glob patterns specified with `*` and `**` match multiple paths.

The `*` pattern matches any file or directory.
It is useful for matching any version directory for versioned documentation.

> The `**` pattern matches all directories and files recursively.

Reviewer 2: I was about to ask for an example in this section and then saw the section above. Thinking we could roll this into the previous section where you end up describing a path anyway.

> The `**` pattern matches all directories and files recursively.

Reviewer 2: Suggest killing this section and just showing a second example using the `**` and an explanation of the outcome.

> Also, if you are adding a new page, consider whether the default labels are appropriate.

Reviewer: ```suggestion
If you add a page, consider whether the default labels are appropriate.
```

    - enterprise
```

#### Set cascading front matter

> When a directory of pages share the same front matter, it is convenient to set them all at once with cascading front matter.

Reviewer: ```suggestion
When a directory of pages shares the same front matter, it is convenient to set them all at once with cascading front matter.
```

There are two forms of cascading front matter.

> The first is a `cascade` map of keys to values.
> The second is a sequence of mappings with the `_target` keyword.

Reviewer 2: ```suggestion

1. The first is a `cascade` map of keys to values.
2. The second is a sequence of mappings with the `_target` keyword.

```

You cannot override the latter with the former.
For more information about cascading front matter, refer to [Front Matter Cascade](https://gohugo.io/content-management/front-matter/#front-matter-cascade).

{{% admonition type="note" %}}
To avoid confusing behavior, set all cascading front matter with the sequence of mappings form.

> {{% /admonition %}}

Reviewer 2: This doesn't seem like it should be a note—seems pretty central to what you're talking about here.

Reviewer: +1

```yaml
cascade:
  - _target:
      path: "/docs/grafana/*/troubleshooting/**"
    labels:
      products:
        - cloud
```

In the preceding example, the `cascade` value is a YAML block sequence of length one, as denoted by the single dash and space (`- `).

The value of the only element in `cascade` sequence is a mapping with two keys: the Hugo reserved `_target`, and the Grafana specific `labels`.

The `_target` key is a map of the key `path` to the value `"/docs/grafana/*/troubleshooting/**"`.
For more information about the the `path` value, refer to [Specify a path](#specify-a-path).

The `labels` key is the same mapping that is used for individual pages.

##### Specify a path

The value of the `_target.path` mapping is a string matching paths from the Hugo content root.
Glob patterns specified with `*` and `**` match multiple paths.

The `*` pattern matches any file or directory.
It is useful for matching any version directory for versioned documentation.

The `**` pattern matches all directories and files recursively.

> 

Reviewer 2: So I vaguely understand this, but this is much more conceptual than instructional and I think it needs to be a bit more instructional. The concepts are probably important, but ultimately someone comes to this to get a job done and maybe it's best to just tell someone "how" rather than "what." Let's sync up on this.

Reviewer: I agree with @imatwawana 's comment. I'll have to study this doc for a bit when needed. Would an instructional video be totally nuts here? Doesn't have to be super polished.
